lib/storage/file.js

```diff
@@ -1060,11 +1069,38 @@ File.prototype.getSignedUrl = function(options, callback) {
     ].join('\n'));
     var signature = sign.sign(credentials.private_key, 'base64');
 
+    var responseContentType = '';
+    if (util.is(options.responseType, 'string')) {
+      responseContentType = 
+        '&response-content-type=' + 
+        encodeURIComponent(options.responseType);
+    }
+
+    var responseContentDisposition = '';
+    if (util.is(options.promptSaveAs, 'string')) {
+      responseContentDisposition =
+        '&response-content-disposition=attachment; filename="' + 
+        encodeURIComponent(options.promptSaveAs) + '"';
+    }
+    if (util.is(options.responseDisposition, 'string')) {
+      if (responseContentDisposition !== '') {
+        console.warn([
+          'getSignedUrl: the promptSaveAs option is ignored',
+          'when responseDisposition is provided.'
```

Comment:
Yeah the only reason someone would set either (or both) of these
parameters is if they read our docs and so we should be ok to assume they
also read the note to go along with it saying that you can only set one or
the other. I say just rip out the console.warn goodies and we're good to go

On Tuesday, May 12, 2015, JJ Geewax <notifications@github.com> wrote:

> In lib/storage/file.js
> <https://github.com/GoogleCloudPlatform/gcloud-node/pull/578#discussion_r30180664>
> :
>
> > +      responseContentType =
> > +        '&response-content-type=' +
> > +        encodeURIComponent(options.responseType);
> > +    }
> > +
> > +    var responseContentDisposition = '';
> > +    if (util.is(options.promptSaveAs, 'string')) {
> > +      responseContentDisposition =
> > +        '&response-content-disposition=attachment; filename="' +
> > +        encodeURIComponent(options.promptSaveAs) + '"';
> > +    }
> > +    if (util.is(options.responseDisposition, 'string')) {
> > +      if (responseContentDisposition !== '') {
> > +        console.warn([
> > +          'getSignedUrl: the promptSaveAs option is ignored',
> > +          'when responseDisposition is provided.'
>
> Is there a way to silently warn...? As in... when you have logging level
> == debug, you see the warning?
>
> If not, I'm fine to silently smash this, but also make it clear in a
> comment that we're blasting out the promptSaveAs value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-node/pull/578/files#r30180664>
> .
>